### ISSUE TITLE
refusing to merge unrelated histories

### ISSUE THREAD

**Issue by BigMichi1** [aormsby/Fork-Sync-With-Upstream-action#37]
suddenly the action stoped working syncing with upstream. at that time I was using version 2.3 of the action,
see here:
https://github.com/BigMichi1/certbot-dns-hetzner/actions/runs/1367119281
https://github.com/BigMichi1/certbot-dns-hetzner/actions/runs/1367244674
between these two executions are 30 minutes and no change has been committed to both repositories at that time

now I tried also upgrading to version 3.1 of the action and now the error is different but still not being able to perform the sync
https://github.com/BigMichi1/certbot-dns-hetzner/runs/4008005789?check_suite_focus=true

the GitHub UI itself does not show anything to merge, also running the commands on command line do not result in one of the above errors

```
git clone git@github.com:BigMichi1/certbot-dns-hetzner.git
cd certbot-dns-hetzner/
git remote add upstream git@github.com:ctrlaltcoop/certbot-dns-hetzner.git
git fetch upstream master
git merge upstream/master
```
shows already up to date

**Comment by aormsby** [comment]
Well, I'm going to need more information about your repos (public/private), what your workflow looks like, and things of that nature to get a better idea of what you're facing.

In the meantime, maybe try v2.4. I made some significant changes in the move to v3, and it's possible that it's not immediately compatible with your setup. However, v2.4 runs with the same flow as v2.3 with some fixes. Worth a shot.

**Comment by aormsby** [comment]
Well, I'm going to have to browse through your work a bit to get a better idea of what you're facing. Not sure I have time right away, though. In the meantime, maybe try v2.4. I made some significant changes in the move to v3, and it's possible that it's not immediately compatible with your setup. However, v2.4 runs with the same flow as v2.3 with some fixes. Worth a shot.

**Comment by aormsby** [comment]
Also try this -- https://github.com/aormsby/Fork-Sync-With-Upstream-action/wiki/Scenarios-for-Advanced-Input#when-your-target-and-upstream-branches-dont-share-a-common-ancestor

`upstream_pull_args: '--allow-unrelated-histories'`

That could be your solution. However, please note there is another bug you might run into detailed here -- #36. I'm working on that next.

**Comment by Wyn-Price** [comment]
Using #38 fixed this for me `uses: aormsby/Fork-Sync-With-Upstream-action@check-fix-2`

**Comment by aormsby** [comment]
#38 merged, released as v3.2
